tests/y-text.tests.js

```diff
@@ -1673,6 +1673,82 @@ export const testDeltaAfterConcurrentFormatting = tc => {
   t.compare(deltas, [[{ retain: 3, attributes: { bold: true } }, { retain: 2, attributes: { bold: null } }]])
 }
 
+/**
+ * Check if the delta is correct after a delete operation. Specifically, we want to make sure that
+ * an embed following the deleted text does not get an attribute update.
+ * 
+ * See https://github.com/yjs/yjs/issues/529
+ * 
+ * @param {t.TestCase} tc
+ */
+export const testDeltaAfterDelete = tc => {
+  const { text0, text1, testConnector } = init(tc, { users: 2 })
+  text0.insert(0, 'abc', {bold: true});
+  text0.insertEmbed(3, new Y.Map([['key', 'val']]));
+  testConnector.flushAllMessages()
+  /**
+   * @type {any}
+   */
+  const deltas = []
+  text1.observe(event => {
+    if (event.delta.length > 0) {
+      deltas.push(event.delta)
+    }
+  })
+  text0.delete(0, 3);
+  testConnector.flushAllMessages()
+  t.compare(deltas, [[{ delete: 3}]])
+}
+
+/**
+ * @param {t.TestCase} tc
+ */
+export const testDeltaAfterDeleteWithEmbedInsideText = tc => {
+  const { text0, text1, testConnector } = init(tc, { users: 2 })
+  const p1 = new Y.XmlText();
+  p1.setAttribute("type", "paragraph")
+  p1.insert(0, "p1")
+  
+  text0.insert(0, "abc", {bold: true});
+  text0.insert(3, "defghi", {bold: false});
+  text0.insertEmbed(6, p1);
+  testConnector.flushAllMessages()
+  /**
+   * @type {any}
+   */
+  const deltas = []
+  text1.observe(event => {
+    if (event.delta.length > 0) {
+      deltas.push(event.delta)
+    }
+  })
+  text0.delete(3, 3);
+  testConnector.flushAllMessages()
+  t.compare(deltas, [[{retain: 3}, { delete: 3}]])
```

Comment:
Before the fix this is `[[{"retain":3},{"delete":3},{"retain":4,"attributes":{"bold":false}}]]`